src/item.cpp

```diff
@@ -4087,19 +4067,13 @@ void item::on_pickup( Character &p )
     if( g->u.getID().is_valid() ) {
         handle_pickup_ownership( p );
     }
-    if( is_bucket_nonempty() ) {
-        contents.spill_contents( p.pos() );
-    }
+    contents.on_pickup( p );
 
     p.flag_encumbrance();
 }
 
 void item::on_contents_changed()
 {
-    if( is_non_resealable_container() ) {
```

Comment:
this is part of that "preserves" wip that i'm actually not sure what it should look like in the end